source/h1_decoder.c

```diff
@@ -714,6 +721,9 @@ static int s_linestate_response(struct aws_h1_decoder *decoder, struct aws_byte_
         return AWS_OP_ERR;
     }
 
+    /* RFC-7230 section 3.3 Message Body */
+    decoder->expects_no_body |= code_val == 304 || code_val == 204 || code_val / 100 == 1;
```

Comment:
304/204/1xx should handle things differently. Check the language of [rfc7230](https://tools.ietf.org/html/rfc7230#section-3.3.2):

> A server MAY send a Content-Length header field in a 304 (Not Modified) response

> A server MUST NOT send a Content-Length header field in any response with a status code of 1xx (Informational) or 204 (No Content).

So Content-Length in a 204/1xx should be an error

PS: I looked at the rules for "Transfer-Encoding" and they're the same as for "Content-Length": MAY be in HEAD or 304 response. MUST NOT be in 1xx or 204 response.

Reply:
Also, add enums to `http_impl.h` for 304 and 204 and use them here